src/exporter.py

```diff
@@ -0,0 +1,196 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Exporter snap helper.
+
+Module focused on handling operations related to prometheus-juju-exporter snap.
+"""
+import logging
+import os
+import subprocess
+from typing import Any, Dict, List, NamedTuple, Optional
+
+import yaml
+from charmhelpers.fetch import snap
+
+# Log messages can be retrieved using juju debug-log
+logger = logging.getLogger(__name__)
+
+
+class ExporterConfigError(Exception):
+    """Indicates problem with configuration of exporter service."""
+
+
+class ExporterConfig(NamedTuple):
+    """Data class that holds information required for exporter configuration."""
+
+    customer: Optional[str] = None
+    cloud: Optional[str] = None
+    controller: Optional[str] = None
+    ca_cert: Optional[str] = None
+    user: Optional[str] = None
+    password: Optional[str] = None
+    interval: Optional[str] = None
+    port: Optional[str] = None
```

Comment:
I think it makes more sense to have these attribute names aligned with [snap config options](https://github.com/canonical/prometheus-juju-exporter/blob/289014b5f22b6c53fecb379feb44b278a36efba7/prometheus_juju_exporter/config_default.yaml) rather than charm. So I'd maybe rename
* `user` -> `username`
* `interval` -> `port_interval`
* `ca_cert` -> `controller_ca_cert`

What do you think?

As for moving validation here. I think it's reasonable enough to have config validation login in `ExporterSnap` class so I'd leave it there.